br/pkg/lightning/restore/restore.go

```diff
@@ -1451,9 +1440,17 @@ func (tr *TableRestore) restoreTable(
 				rowIDMax = engine.Chunks[len(engine.Chunks)-1].Chunk.RowIDMax
 			}
 		}
+		db, _ := rc.tidbGlue.GetDB()
+		versionStr, err := version.FetchVersion(ctx, db)
+		if err != nil {
+			return false, errors.Trace(err)
+		}
+
+		versionInfo := version.ParseServerInfo(versionStr)
 
 		// "show table next_row_id" is only available after v4.0.0
-		if tidbVersion.Major >= 4 && (rc.cfg.TikvImporter.Backend == config.BackendLocal || rc.cfg.TikvImporter.Backend == config.BackendImporter) {
+		if versionInfo.ServerType == version.ServerTypeTiDB && versionInfo.ServerVersion.Major >= 4 &&
```

Comment:
It seems that if ServerType does not equal `ServerTypeTiDB`, this backend can not be `BackendLocal` and `BackendImporter`.  We can check it in config